ang/test/civiawards/directives/award.directive.spec.js

```diff
@@ -0,0 +1,255 @@
+/* eslint-env jasmine */
+(function (_) {
+  describe('civiaward', () => {
+    var $q, $controller, $rootScope, $scope, $window, crmApi,
+      CaseStatus, AwardData, AwardAdditionalDetailsData;
+
+    beforeEach(module('civicase-base', 'civiawards.templates', 'civiawards', 'civicase.data', 'civiawards.data', ($provide) => {
+      $provide.value('crmApi', jasmine.createSpy('crmApi'));
+      $provide.value('$window', { location: {} });
+    }));
+
+    beforeEach(inject((_$q_, _$controller_, _$window_, _$rootScope_, _crmApi_,
+      _CaseStatus_, _AwardData_, _AwardAdditionalDetailsData_) => {
+      $q = _$q_;
+      $window = _$window_;
+      $controller = _$controller_;
+      $rootScope = _$rootScope_;
+      crmApi = _crmApi_;
+      CaseStatus = _CaseStatus_;
+      AwardData = _AwardData_;
+      AwardAdditionalDetailsData = _AwardAdditionalDetailsData_;
+      $scope = $rootScope.$new();
+
+      crmApi.and.returnValue($q.resolve({}));
+      $scope.$digest();
+      $scope.basic_details_form = { awardName: { $pristine: true } };
+    }));
+
+    describe('basic tests', () => {
+      describe('when creating new award', function () {
+        beforeEach(() => {
+          createController({ ifNewAward: true });
+        });
+
+        it('name field is disabled', () => {
+          expect($scope.isNameDisabled).toBe(true);
+        });
+
+        it('submit button is disabled', () => {
+          expect($scope.submitInProgress).toBe(false);
+        });
+
+        it('shows all award stages', () => {
+          expect($scope.awardStages).toBe(CaseStatus.getAll());
+        });
+
+        describe('all basic fields', () => {
+          var expectedSelectedAwardStages = {};
+
+          beforeEach(function () {
+            _.each(CaseStatus.getAll(), (awardStage) => {
+              expectedSelectedAwardStages[awardStage.value] = true;
+            });
+          });
+
+          it('all basic details fields are empty', () => {
+            expect($scope.basicDetails).toEqual({
+              title: '',
+              name: '',
+              description: '',
+              awardType: null,
+              isEnabled: true,
+              startDate: null,
+              endDate: null,
+              awardManagers: [],
+              selectedAwardStages: expectedSelectedAwardStages
+            });
+          });
+        });
+
+        describe('name field', () => {
+          beforeEach(() => {
+            $scope.basicDetails.title = 'Some custom text';
+            $scope.$digest();
+          });
+
+          it('updating the title changes the name field too', () => {
+            expect($scope.basicDetails.name).toBe('some_custom_text');
+          });
```

Comment:
```js
describe('when the title changes', () => {
  beforeEach(() => {
    $scope.basicDetails.title = 'Some custom text';
    $scope.$digest();
  });

  it('updating the title changes the name field too', () => {
    expect($scope.basicDetails.name).toBe('some_custom_text');
  });
});

// ....

describe('when the name is manually changed', () => {
  // ...
});
```

Here you show that there are two things that happen. One when the title changes, the other when the name changes.